magda-gateway/src/index.ts

```diff
@@ -333,6 +345,9 @@ app.use("/", createGenericProxy(argv.web));
 app.listen(argv.listenPort);
 console.log("Listening on port " + argv.listenPort);
 
+export let tenantsTable = new Map<String, Tenant>();
+loadTenantsTable(tenantsTable, `${argv.externalUrl}/api/v0/registry`);
```

Comment:
I think we'll need a way to refresh this automatically, not just load it at startup. I'd like to say that can be a problem for later, but I don't think it actually can be, unfortunately.

Reply:
@mwu2018 Maybe look at the way the web server pulls in stuff from a content api, I think that pattern works pretty good for this - it uses lodash debounce to memoize the results in such a way that they still get refreshed every so often.

Reply:
Revised.
As the first step to multi-tenant support, we use default tenant, which is single tenant by default. In the future, after magda admin portal user adds a new tenant, the tenants table needs to be refreshed via API that has been added but disabled by default.